src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -8701,6 +8736,72 @@ def update_azure_container_storage(self, mc: ManagedCluster) -> ManagedCluster:
 
         return mc
 
+    def update_azure_container_storage_v2(self, mc: ManagedCluster) -> ManagedCluster:
+        self._ensure_mc(mc)
+        enable_azure_container_storage_v2 = self.context.raw_param.get("enable_azure_container_storage_v2")
+        disable_azure_container_storage_v2 = self.context.raw_param.get("disable_azure_container_storage_v2")
+        storage_pool_name = self.context.raw_param.get("storage_pool_name")
+        pool_sku = self.context.raw_param.get("storage_pool_sku")
+        pool_option = self.context.raw_param.get("storage_pool_option")
+        pool_size = self.context.raw_param.get("storage_pool_size")
+
+        if enable_azure_container_storage_v2 and disable_azure_container_storage_v2:
+            raise MutuallyExclusiveArgumentError(
+                'Conflicting flags. Cannot set --enable-azure-container-storage-v2 '
+                'and --disable-azure-container-storage-v2 together.'
+            )
+
+        if enable_azure_container_storage_v2 or disable_azure_container_storage_v2:
+            from azure.cli.command_modules.acs.azurecontainerstorage._helpers import get_container_storage_v2_extension_installed
+            is_extension_installed, _ = get_container_storage_v2_extension_installed(
+                self.cmd,
+                self.context.get_resource_group_name(),
+                self.context.get_name()
+            )
+
+            from azure.cli.command_modules.acs.azurecontainerstorage._helpers import get_container_storage_v1_extension_installed
+            is_containerstorage_v1_installed, v1_extension_version = get_container_storage_v1_extension_installed(
+                self.cmd,
+                self.context.get_resource_group_name(),
+                self.context.get_name()
+            )
+
+        if enable_azure_container_storage_v2:
+            from azure.cli.command_modules.acs.azurecontainerstorage._validators import (
+                validate_enable_azure_container_storage_v2_params,
+            )
+            validate_enable_azure_container_storage_v2_params(
+                is_extension_installed,
+                is_containerstorage_v1_installed,
+                v1_extension_version,
+                storage_pool_name,
+                pool_sku,
+                pool_option,
+                pool_size,
+            )
+
+        if disable_azure_container_storage_v2:
+            msg = (
+                "Please make sure there are no existing PVs and PVCs that are provisioned by Azure Container Storage before disabling. "
```

Comment:
This error needs to explicitly say that nodes could be left in unclean state. We should also add an explicit request to consent, i.e. 

```
Please ensure there are no existing PersistentVolumes (PVs) or PersistentVolumeClaims (PVCs) provisioned by Azure Container Storage before proceeding.
If you disable Azure Container Storage V2 while PVs or PVCs remain, they can only be cleaned up after re-enabling it using:
az aks update --enable-azure-container-storage-v2
Note: This action may leave nodes in an unclean state, and any local data associated with the volumes will not be erased.

Do you want to continue? [y/N]
```

Reply:
Added the explicit statement about unclean nodes. The prompt is being raised by line 8812.

Reply:
The updated message now only mentions that data will be left on the node but we should also say that nodes will stay in an unclean state as all lvm resources will also be leaked.